### ISSUE TITLE
Gir: Improve async method generation

### ISSUE THREAD

**Issue by badcel** [gircore/gir.core#429]
There is some work happening for the gir generation to add explicit information for async methods.

The work is not yet done. Once it is finished support the new annotations:

Issue: https://gitlab.gnome.org/GNOME/gobject-introspection/-/issues/28

PR: https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/278

**Comment by badcel** [comment]
Superseded by #893.
